torch/_prims/rng_prims.py

```diff
@@ -70,7 +70,7 @@ def philox_rand_offset(
     device_property = torch.cuda.get_device_properties(torch.cuda.current_device())
     blocks_per_sm = device_property.max_threads_per_multi_processor // block_size
     grid_size = (numel + block_size - 1) // block_size
-    grid_size = min(grid_size, device_property.multi_processor_count * blocks_per_sm)
+    grid_size = min(grid_size, device_property.multi_processor_count * blocks_per_sm)  # type: ignore[call-overload]
```

Comment:
I think this is out of the scope of this PR as I'm not sure why we use `Tensor` (instead of `int`) for `grid_size` here.

The error is no overload for `min(Tensor, int)`. This is because `Tensor.__lt__()` returns `Tensor`, incompatible with typeshed's annotation on `min`, which requires:
https://github.com/python/typeshed/blob/052d2b9f3a9afaa56b86f0c84bc9d8769458d96a/stdlib/_typeshed/__init__.pyi#L53-L54
This requirement is introduced in python/typeshed#7093, where they ask `__lt__` must return `bool` but not any other bool-compatible type.